include/lsst/meas/extensions/photometryKron/KronFluxAlgorithm.h

```diff
@@ -0,0 +1,130 @@
+                       "if true, use existing shape and centroid measurements instead of fitting");
+    LSST_CONTROL_FIELD(nSigmaForRadius, double,
+                       "Multiplier of rms size for aperture used to initially estimate the Kron radius");
+    LSST_CONTROL_FIELD(nIterForRadius, int, "Number of times to iterate when setting the Kron radius");
+    LSST_CONTROL_FIELD(nRadiusForFlux, double, "Number of Kron radii for Kron flux");
+    LSST_CONTROL_FIELD(maxSincRadius, double,
+                       "Largest aperture for which to use the slow, accurate, sinc aperture code");
+    LSST_CONTROL_FIELD(minimumRadius, double,
+                       "Minimum Kron radius (if == 0.0 use PSF's Kron radius). "
+                       "Ignored if enforceMinimumRadius is false");
+    LSST_CONTROL_FIELD(enforceMinimumRadius, bool, "If true check that the Kron radius exceeds some minimum");
+    LSST_CONTROL_FIELD(useFootprintRadius, bool,
+                       "Use the Footprint size as part of initial estimate of Kron radius");
+    LSST_CONTROL_FIELD(smoothingSigma, double,
+                       "Smooth image with N(0, smoothingSigma^2) Gaussian while estimating R_K");
+    KronFluxControl() :
+        fixed(false),
+        nSigmaForRadius(6.0),
+        nIterForRadius(1),
+        nRadiusForFlux(2.5),
+        maxSincRadius(10.0),
+        minimumRadius(0.0),
+        enforceMinimumRadius(true),
+        useFootprintRadius(false),
+        smoothingSigma(-1.0)
+    {}
+};
+/**
+ *  @brief A measurement algorithm that estimates flux using Kron photometry
+ */
+class KronFluxAlgorithm : public base::SimpleAlgorithm {
+public:
+
+    enum {
+        FAILURE=base::FlagHandler::FAILURE,
+        RADIUS,
+        SMALL_RADIUS,
+        N_FLAGS
+    };
+
+    /// A typedef to the Control object for this algorithm, defined above.
+    /// The control object contains the configuration parameters for this algorithm.
+    typedef KronFluxControl Control;
+
+    KronFluxAlgorithm(Control const & ctrl, std::string const & name, afw::table::Schema & schema);
+
+private:
+
+    virtual void measure(
+        afw::table::SourceRecord & measRecord,
+        afw::image::Exposure<float> const & exposure
+    ) const;
+
+    virtual void fail(
+        afw::table::SourceRecord & measRecord,
+        meas::base::MeasurementError * error=NULL
+    ) const;
+
+
+    void _applyAperture(
+        afw::table::SourceRecord & source,
+        afw::image::Exposure<float> const& exposure,
+        KronAperture const& aperture
+        ) const;
+
+    void _applyForced(
+        afw::table::SourceRecord & source,
+        afw::image::Exposure<float> const & exposure,
+        afw::geom::Point2D const & center,
+        afw::table::SourceRecord const & reference,
+        afw::geom::AffineTransform const & refToMeas
+    ) const;
+    std::string const & _name;
+    Control _ctrl;
+    //algorithms::ScaledFlux::KeyTuple _fluxCorrectionKeys;
```

Comment:
OK, that's perfectly reasonable.  I was trying to save time, not add
confusion.  With many of the code conversions I have done, the code cannot
be converted for one reason or another.  I will leave the old code in where
it seems useful, and add a descriptive comment where it is not.

On Wed, Feb 18, 2015 at 5:43 PM, John Swinbank <notifications@github.com>
wrote:

> In include/lsst/meas/extensions/photometryKron/KronFluxAlgorithm.h
> <https://github.com/lsst/meas_extensions_photometryKron/pull/1#discussion_r24960162>
> :
>
> > +    void _applyAperture(
> > +        afw::table::SourceRecord & source,
> > +        afw::image::Exposure<float> const& exposure,
> > +        KronAperture const& aperture
> > +        ) const;
> > +
> > +    void _applyForced(
> > +        afw::table::SourceRecord & source,
> > +        afw::image::Exposure<float> const & exposure,
> > +        afw::geom::Point2D const & center,
> > +        afw::table::SourceRecord const & reference,
> > +        afw::geom::AffineTransform const & refToMeas
> > +    ) const;
> > +    std::string const & _name;
> > +    Control _ctrl;
> > +    //algorithms::ScaledFlux::KeyTuple _fluxCorrectionKeys;
>
> I think that
>
> saying that I should never have commented code, even for a good reason
>
> is not accurate: if there's a good reason, then of course you should do
> it. But, in general, a good reason is pretty hard to find.
>
> Given that the whole history of the codebase is stored in git and that
> commented code like this adds potentially confusing noise for anybody who
> isn't trying to (re-)implement the feature in question, I think it should
> be gone (with a note in the JIRA ticket requesting the reimplementation
> which points out where it can be found). If you really, really think it's
> necessary and that the git history is inadequate, you should make sure the
> commented code is clearly annotated to say what it's for and why it's
> required.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lsst/meas_extensions_photometryKron/pull/1/files#r24960162>
> .
>